projects/applicationinsights-angularplugin-js/src/lib/applicationinsights-angularplugin-js.component.ts

```diff
@@ -37,83 +46,133 @@ export class AngularPlugin extends BaseTelemetryPlugin {
+        let _prevRouter: Router;
+
+        dynamicProto(AngularPlugin, this, (_self, _base) => {
+            _initDefaults();
+
+            _self.initialize = (config: IConfiguration & IConfig, core: IAppInsightsCore, extensions: IPlugin[], pluginChain?: ITelemetryPluginChain) => {
+                super.initialize(config, core, extensions, pluginChain);
+        
+                _self._addHook(onConfigChange(config, (details) => {
+                    let ctx = _self._getTelCtx();
+                    _angularCfg = ctx.getExtCfg<IAngularExtensionConfig>(_self.identifier, defaultAngularExtensionConfig);
+                    _propertiesPlugin = core.getPlugin<properties.PropertiesPlugin>(PropertiesPluginIdentifier)?.plugin as properties.PropertiesPlugin;
+                    _analyticsPlugin = core.getPlugin<AnalyticsPlugin>(AnalyticsPluginIdentifier)?.plugin as AnalyticsPlugin;
+
+                    if (_analyticsPlugin) {
+                        if (ApplicationinsightsAngularpluginErrorService.instance !== null) {
+                            ApplicationinsightsAngularpluginErrorService.instance.plugin = _analyticsPlugin;
+                            if (_angularCfg.errorServices && Array.isArray(_angularCfg.errorServices)) {
+                                ApplicationinsightsAngularpluginErrorService.instance.clearErrorHandlers();
+                                // another method would be use blkVal, and compare the two array value to decide  
+                                // whether clear it or not, but it will cause more performance issue.
+                                arrForEach(_angularCfg.errorServices, (errorService: IErrorService) => {
+                                    ApplicationinsightsAngularpluginErrorService.instance.addErrorHandler(errorService);
+                                });
+                            }
+                        }
                     }
-                }
-            }
+            
+                    if (_angularCfg.router !== _prevRouter) {
+                        // router is changed, or has not been initialized yet
 
-            if (self._eventSubscription) {
-                self._eventSubscription.unsubscribe();
-                self._eventSubscription = null;
-            }
+                        // unsubscribe previous router events
+                        if (_eventSubscription) {
+                            _eventSubscription.unsubscribe();
+                        }
 
-            self.analyticsPlugin = null;
-            self.propertiesPlugin = null;
-            self._angularCfg = null;
-        }
-    }
+                        if (_angularCfg.router){
+                            // subscribe to new router events
+                            const pageViewTelemetry: IPageViewTelemetry = {
+                                uri: _angularCfg.router.url
+                            };
+                            _self.trackPageView(pageViewTelemetry);
```

Comment:
Will need to "protect" this so we don't trigger a page view for every config change.

ie. Only send a page view for the first "initialize"